### PR TITLE
[Torch Op] Add AtenChunkOp support

### PR DESCRIPTION
**Pull request by zhekunz2**
Recompose `AtenChunkOp` + `PrimListUnpackOp` to `AtenSliceTensorOp`s.
Recompose rule is equivalent to the following
```
input = chunkOp.getSelf()
chunks = chunkOp.getChunks()
dim = chunkOp.getDim()
totalSize = torch.size(input, dim)
if floordiv(totalSize, chunks) * chunks == totalSize:
    chunkSize = floordiv(totalSize, chunks)
else:
    chunkSize = floordiv(totalSize, chunks) + 1

slices = []
for i in range(primList.numResults()):
    start = chunkSize * i
    if i == primList.numResults() - 1:
        end = totalSize - start
    else:
        end = chunkSize * (i+1)
    slices.append(torch.ops.aten.slice(input, dim, start, end, /*step*/1)

replaceOp(primList, slices)
```

### CODE REVIEW

File: lib/Dialect/Torch/Transforms/RecomposeComplexOps.cpp [hunk 1]
```diff
@@ -223,6 +223,81 @@ class RecomposeSplitTensorGetItemOp
...
+      return failure();
+    Value dim = chunk.getDim();
+    Value input = chunk.getSelf();
+    Value chunks = chunk.getChunks();
+    Location loc = chunk.getLoc();
+    Value totalSize = rewriter.create<Torch::AtenSizeIntOp>(loc, input, dim);
+
+    // below code is equivalent to:
+    // if floordiv(totalSize, chunks) * chunks == totalSize:
+    //    chunkSize = floordiv(totalSize, chunks)
```
Comment (qingyunqu) [inline lib/Dialect/Torch/Transforms/RecomposeComplexOps.cpp:246]: I think it's no need to insert `PrimIfOp`, `chunkSize = floordiv(totalSize + chunks - 1, chunks)` is more simple.

**Review by qingyunqu — CHANGES_REQUESTED** [review]

**Review by qingyunqu — APPROVED** [review]

**Comment by qingyunqu** [comment]
LGTM

File: lib/Dialect/Torch/Transforms/RecomposeComplexOps.cpp [hunk 2]
```diff
@@ -223,6 +223,58 @@ class RecomposeSplitTensorGetItemOp
...
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(PrimListUnpackOp op,
+                                PatternRewriter &rewriter) const override {
+    // recompose AtenChunkOp + PrimListUnpackOp to AtenSliceTensorOps
+    auto chunk = dyn_cast<AtenChunkOp>(op.getOperand().getDefiningOp());
+    if (!chunk)
+      return failure();
+    if (isListPotentiallyMutated(chunk.getResult()))
+      return failure();
```
Comment (ramiro050) [inline lib/Dialect/Torch/Transforms/RecomposeComplexOps.cpp:237]: nit: can you use `rewriter.notifyMatchFailure(...)` with an error message? Makes it easier to debug from `-debug` logs

**Review by ramiro050 — APPROVED** [review]
Just one small comment, other than that LGTM

**Pull request merged**